### ISSUE TITLE
[stable/datadog] system-probe fails to start

### ISSUE THREAD

**Issue by yafernandes** [helm/charts#21348]
Kubernetes 1.17.3
Ubuntu 19.10
Kernel 5.3.0
Docker 19.03.6

system-probe fails to start with error 128.

Removing the line below from the DaemonSet fixes the issue.
`container.seccomp.security.alpha.kubernetes.io/system-probe: localhost/system-probe`

values.yaml
```
datadog:
  apiKeyExistingSecret: dd
  apm:
    enabled: true
  logs:
    enabled: true
    containerCollectAll: true
  processAgent:
    enabled: true
  systemProbe:
    enabled: true
  clusterChecks:
    enabled: true
  kubeStateMetricsEnabled: true
  dogstatsd:
    nonLocalTraffic: true
  env:
    - name: DD_HTTP_PROXY
      value: http://proxy.k8s.aws.pipsquack.ca:3128/
    - name: DD_HTTPS_PROXY
      value: http://proxy.k8s.aws.pipsquack.ca:3128/
  confd:
    kube_controller_manager.yaml: |-
      ad_identifiers:
        - kube-controller-manager
      init_config:
      instances:
        - prometheus_url: http://%%host%%:10252/metrics
    kube_scheduler.yaml: |-
      ad_identifiers:
        - kube-scheduler
      init_config:
      instances:
        - prometheus_url: http://%%host%%:10251/metrics
    etcd.yaml: |-
      ad_identifiers:
        - etcd
      init_config:
      instances:
        - use_preview: true
          prometheus_url: https://%%host%%:2379/metrics
          ssl_ca_cert: /etc/kubernetes/pki/etcd/ca.crt
          ssl_cert: /etc/kubernetes/pki/etcd/peer.crt
          ssl_private_key: /etc/kubernetes/pki/etcd/peer.key
    nginx_ingress_controller.yaml: |-
      ad_identifiers:
        - nginx-ingress-controller
      init_config:
      instances:
        - prometheus_url: http://%%host%%:10254/metrics
    nginx.yaml: |-
      ad_identifiers:
        - nginx-ingress-controller
      init_config:
      instances:
        - nginx_status_url: http://%%host%%:18080/metrics
    kubelet.yaml: |-
      init_config:
      instances:
        - 
agents:
  enabled: true
  rbac:
    create: true
  volumeMounts:
    - mountPath: /etc/kubernetes/pki/etcd
      name: etcd-certs
    - mountPath: /etc/datadog-agent/conf.d/etcd.d
      name: empty-dir
    - mountPath: /etc/datadog-agent/conf.d/kubelet.d
      name: empty-dir
  volumes:
    - name: etcd-certs
      hostPath:
        path: /etc/kubernetes/pki/etcd
        type: DirectoryOrCreate
    - name: empty-dir
      emptyDir: {}
  tolerations:
    - operator: Exists
clusterAgent:
  enabled: true
  rbac:
    create: true
```

**Comment by clamoriniere** [comment]
Hi @yafernandes 

I understood you issue: it is because you are mounting an `empty-dir` at the location of the default `kubelet` conf.d folder.

```
    - mountPath: /etc/datadog-agent/conf.d/kubelet.d
      name: empty-dir
```
You don't need to do that if you want to overwrite after the kubelet config thanks to `datadog.confd` values.yaml parameter.

Please let me know if it fix the issue.

**Comment by clamoriniere** [comment]
Hi @yafernandes 
it seems link to your kernel version and the `seccomp` profile provide in the helm chart.

Could you please add the option `datadog.systemProbe.seccomp: "runtime/default"` in your `values.yaml` file to validate that it is the `seccomp` profile that is too restrictive.

**Comment by rslinckx** [comment]
Using runtime/default fixed the issue for me, with 2.0.11 chart. Also had the system-probe crashing using GKE 1.15 with cos image.

**Comment by L3n41c** [comment]
Hi @yafernandes,
I’ve created #21526 with the fix that we validated together.
